### PR TITLE
#269 Collapsible panels

### LINKED ISSUE #269: collapsible "Modules" and "Exercises"
so the idea is quite simple, when 2-clicking on the e.g. "Modules" subtab, it collapses and gives all the space to "Exercises"
before:
before_collapse
after (but without that scope behind the red "zig-zagged" line in between)
after_collapse

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Closes #269

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [ ] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [ ] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [ ] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

**Review by stellatsv — APPROVED** [review]
Good work! Please remember to remove the extra Modules view before merging :wink:

File: src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java [hunk 1]
```diff
@@ -218,6 +221,31 @@ public void setAssistantMode(boolean assistantMode) {
             String.valueOf(assistantMode));
   }
 
+  /**
+   * Sets a collapsible panel collapsed.
+   */
+  public void setCollapsed(@NotNull String title) {
+    applicationPropertiesManager.setValue(A_PLUS_COLLAPSED_PANELS.getName(),
+        getCollapsed() != null ? getCollapsed() + ";" + title : title);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java:229]: Remark: this kind of logic could be handled in a separate (domain oblivious) class (that ideally would implement `Set<String>` interface). That could then be easily unit-tested. Just consider! :)

File: src/main/java/fi/aalto/cs/apluscourses/intellij/toolwindows/APlusToolWindowFactory.java [hunk 2]
```diff
@@ -31,26 +31,28 @@
   @Override
   protected JComponent createToolWindowContentInternal(@NotNull Project project) {
     ModulesView modulesView = createModulesView(project);
+    // TODO remove, for demo purposes
+    ModulesView modulesView2 = createModulesView(project);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/toolwindows/APlusToolWindowFactory.java:35]: Remove

**Review by OlliKiljunen — COMMENTED** [review]
Very good. Just remove lines that were marked for removal and this is good to go!

**Pull request merged**